cmd/bastion-tunnel/cred.go

```diff
@@ -22,9 +24,18 @@ func (s *staticTokenCredential) GetToken(ctx context.Context, options policy.Tok
 	}, nil
 }
 
-func createCred(fixedtoken string) (azcore.TokenCredential, error) {
+func createCred(fixedtoken string, opts *azcore.ClientOptions) (azcore.TokenCredential, error) {
 	var creds []azcore.TokenCredential
 
+	// From azure sdk for go docs:
+	//    DisableInstanceDiscovery should be set true only by applications authenticating in
+	//    disconnected clouds, or private clouds such as Azure Stack.
+	disableInstanceDiscovery := false
+	if !(reflect.DeepEqual(opts.Cloud, cloud.AzurePublic) || reflect.DeepEqual(opts.Cloud, cloud.AzureGovernment) || reflect.DeepEqual(opts.Cloud, cloud.AzureChina)) {
+		disableInstanceDiscovery = true
+	}
+
+	// Add more credentials to the array here if needed
```

Comment:
does cli cred need this opt?

Reply:
Based on the docs, I don't believe so: [azidentity#NewAzureCLICredential](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#NewAzureCLICredential).

My best guess is that since AzCli simply grabs your login information from the active session the go sdk doesn't care. If you needed to log into a different cloud, you would do that with these commands [Azure cloud management with the Azure CLI](https://learn.microsoft.com/en-us/cli/azure/manage-clouds-azure-cli). The go library would then just grab the cloud specific token from AzCli